internal/controlplane/handlers_authz.go

```diff
@@ -148,62 +125,21 @@ func AuthorizedOnProject(ctx context.Context, projectID uuid.UUID) error {
 	return nil
 }
 
-// AuthUnaryInterceptor is a server interceptor for authentication
-func AuthUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
+// AuthorizationUnaryInterceptor is a server interceptor that sets up the user permissions
+func AuthorizationUnaryInterceptor(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler) (any, error) {
 
-	opts, err := optionsForMethod(info)
-	if err != nil {
-		// Fail closed safely, rather than log and proceed.
-		return nil, status.Errorf(codes.Internal, "Error getting options for method: %v", err)
-	}
-
-	ctx = withRpcOptions(ctx, opts)
-
-	if opts.GetAnonymous() {
-		if !opts.GetNoLog() {
-			zerolog.Ctx(ctx).Info().Msgf("Bypassing authentication")
-		}
-		return handler(ctx, req)
-	}
-
-	token, err := gauth.AuthFromMD(ctx, "bearer")
-	if err != nil {
-		if statusErr, ok := status.FromError(err); ok {
-			return nil, util.FromRpcError(statusErr)
-		}
-		return nil, status.Errorf(codes.Unauthenticated, "no auth token: %v", err)
-	}
-
 	server := info.Server.(*Server)
 
-	parsedToken, err := server.vldtr.ParseAndValidate(token)
-	if err != nil {
-		return nil, status.Errorf(codes.Unauthenticated, "invalid auth token: %v", err)
+	subject := auth.GetUserSubjectFromContext(ctx)
+	if subject == nil {
+		return nil, status.Errorf(codes.Internal, "Error getting user subject")
```

Comment:
Won't this always fail for methods marked as `anonymous`, because they don't store a subject?

Reply:
I moved this call to `lookupUserPermissions`, so now anonymous methods won't error, but will simply have empty permissions.